src/pages/create-product/index.js

```diff
@@ -0,0 +1,170 @@
+import BaseComponent from "../../components/base-component.js";
+import { createProduct, getBrands, getCategories } from "../../api/products.js";
+
+import "./create-product.css";
+
+export default class CreateProductPage extends BaseComponent {
+  constructor() {
+    super();
+    this.createProduct = createProduct;
+    this.getCategories = getCategories;
+    this.getBrands = getBrands;
+
+    this.init();
+    this.initEventListeners();
+    this.loadData();
+  }
+
+  async loadData() {
+    try {
+      const [categories, brands] = await Promise.all([
+        this.getCategories(),
+        this.getBrands(),
+      ]);
+
+      this.update(categories, brands);
+    } catch (error) {
+      this.showAlert("error", error.message);
+    }
+  }
+
+  update(categories = [], brands = []) {
+    this.updateOptions("brandsSelect", brands);
+    this.updateOptions("categoriesSelect", categories);
+
+    this.subElements.fieldset.removeAttribute("disabled");
+  }
+
+  updateOptions(name = "", data = []) {
+    const fragment = new DocumentFragment();
+
+    for (const brand of data) {
+      const key = brand.toLowerCase().split(" ").join("_");
+      const option = new Option(brand, key);
+      fragment.append(option);
+    }
+
+    this.subElements[name].replaceChildren(fragment);
+  }
+
+  startLoading() {
+    this.subElements.fieldset.setAttribute("disabled", "true");
+    this.subElements.submitBtn.innerHTML = `
+      <span class="spinner-border spinner-border-sm" aria-hidden="true"></span>
+      <span role="status">Loading...</span>
+    `;
+  }
+
+  stopLoading() {
+    this.subElements.fieldset.removeAttribute("disabled");
+    this.subElements.submitBtn.innerHTML = "Submit";
+  }
+
+  initEventListeners() {
+    const formElement = this.subElements.form;
+
+    formElement.addEventListener("submit", async (event) => {
+      event.preventDefault();
+
+      if (!formElement.checkValidity()) {
+        formElement.classList.add("was-validated");
+        return;
+      }
+
+      const formData = new FormData(formElement);
+
+      try {
+        this.startLoading();
+        await this.createProduct(formData);
+
+        this.showAlert("success", "Product was successfully created.");
+        formElement.reset();
+      } catch (error) {
+        this.showAlert("error", "Error during product creation.");
+        this.showValidationErrors(error);
+      } finally {
+        formElement.classList.remove("was-validated");
+        this.stopLoading();
+      }
+    });
```

Comment:
**Scalability Issue**: The current implementation of form submission handling within a single event listener could lead to scalability issues, as it handles multiple aspects like validation, loading state changes, and API requests sequentially. <br> **Fix**: Refactor the form submission to use asynchronous processing and separate concerns. This can include breaking down the function into smaller functions handling specific tasks and using async/await more effectively to manage state transitions and error handling. <br> **Code Suggestion**: 
 ```
 +    async function handleFormSubmission(event) {
 +      event.preventDefault();
 +      if (!formElement.checkValidity()) {
 +        formElement.classList.add("was-validated");
 +        return;
 +      }
 +      this.startLoading();
 +      try {
 +        await this.createProduct(new FormData(formElement));
 +        this.showAlert("success", "Product was successfully created.");
 +        formElement.reset();
 +      } catch (error) {
 +        this.showAlert("error", "Error during product creation.");
 +        this.showValidationErrors(error);
 +      } finally {
 +        this.stopLoading();
 +        formElement.classList.remove("was-validated");
 +      }
 +    }
 +    formElement.addEventListener("submit", handleFormSubmission.bind(this));
 ```